rslearn/dataset/dataset.py

```diff
@@ -48,10 +48,13 @@ def __init__(self, path: UPath) -> None:
         # Load dataset configuration.
         with (self.path / "config.json").open("r") as f:
             config = json.load(f)
-            self.layers: dict = {
-                layer_name: load_layer_config(d)
-                for layer_name, d in config["layers"].items()
-            }
+            self.layers = {}
+            for layer_name, d in config["layers"].items():
+                # Layer names must not contain period, since we use period to
+                # distinguish different materialized groups within a layer.
+                assert "." not in layer_name
```

Comment:
It will be better to add a message to this assert statement, like what's described in the comment

Reply:
thanks, added!